src/main/java/com/project/escape/domain/reservation/ReservationService.java

```diff
@@ -23,6 +23,11 @@ public class ReservationService {
     private final ReservationRepository reservationRepository;
     private final UserReservationRepository userReservationRepository;
 
+    @Transactional
+    public void saveAllReservations(List<Reservation> reservations) {
+        reservationRepository.saveAll(reservations);
```

Comment:
ALL 류의 쿼리들은 늘 조심해야 할것 같아요. 성능에 직결되니..

Reply:
```suggestion
       reservations.forEach(reservationRepository::save);
```

이렇게 변경하면 아래처럼 소나린트가 추천을 해줍니다. 이 권고는 무시해도 되는 것이겠죠?
<img width="541" alt="image" src="https://github.com/f-lab-edu/escape/assets/58449849/f808fa45-36a9-4de3-8e70-9029c97c3e1d">

Reply:
`saveAll()`이 성능상 좋지 않을 수 있는지 찾아보았습니다.
---
일반적으로 JPA는 영속성 컨텍스트에 저장된 엔티티 객체의 상태 변화를 추적하여 변경된 부분만 UPDATE 쿼리를 생성합니다. 하지만 `saveAll()`을 사용하면 JPA는 저장하려는 모든 엔티티를 영속 상태로 변환하려고 시도합니다.

이때 만약 해당 엔티티가 이미 영속성 컨텍스트에 존재한다면, JPA는 불필요한 UPDATE 쿼리를 생성할 수 있습니다. 이는 성능 저하의 원인이 될 수 있습니다.

따라서 대량의 INSERT 작업에는 `saveAll()`이 효율적이지만, UPDATE 작업에는 `save()`를 개별적으로 사용하는 것이 더 나은 선택일 수 있습니다.

`saveAll()`과 같이 대량의 엔티티를 다룰 때, 불필요한 쿼리 생성으로 인해 성능 이슈가 발생할 수 있다는 의미입니다. 따라서 상황에 따라 적절한 메서드를 선택하는 것이 중요합니다.

Reply:
JPA에서 `saveAll()`과 `save()`의 성능을 비교하면 다음과 같습니다.
---

#### `saveAll()`

- 여러 개의 엔티티를 한 번에 저장할 수 있습니다.
- 한 번의 트랜잭션으로 처리되므로 성능이 좋습니다.
- 영속성 컨텍스트 관리가 효율적입니다.
- 단점은 한 엔티티에서 예외가 발생하면 전체 작업이 실패합니다.

#### `save()`

- 하나의 엔티티를 저장합니다.
- 매번 새로운 트랜잭션이 열리므로 오버헤드가 있습니다.
- 개별 엔티티에 대한 예외 처리가 용이합니다.
- 일반적으로 대량의 데이터를 한번에 저장해야 한다면 `saveAll()`을 사용하는 것이 효율적입니다. 하지만 개별 - 엔티티에 대한 예외 처리가 중요하다면`save()`를 사용하는 것이 좋습니다.

성능 측면에서는 `saveAll()`이 `save()`보다 월등히 좋습니다. 한 번의 트랜잭션으로 처리되고 영속성 컨텍스트 관리도 효율적이기 때문입니다. 하지만 데이터의 크기와 예외 처리 요구사항에 따라 적절히 판단해야 합니다.

Reply:
select -> 몇 백만건이 들어올 것을 생각하여 페이지네이션 하기
insert -> 배치를 사용하여 chunk단위로 나눠서 하기.